systemd/crc-no-tap.service

```diff
@@ -7,7 +7,8 @@ RequiresMountsFor=/etc/NetworkManager/system-connections
 
 [Service]
 Type=oneshot
-ExecStart=/usr/bin/rm -f /etc/NetworkManager/system-connections/tap0.nmconnection
+EnvironmentFile=-/etc/sysconfig/crc-env
+ExecStart=/usr/local/bin/crc-no-tap.sh
```

Comment:
maybe we should have two versions of this service, one with 

`ConditionVirtualization=apple` and the other with `ExecCondition=/usr/local/bin/check-crc-ng-env.sh`

the first removes this file in case we are running on macOS, and the other service will remove it when `CRC_CLOUD` or `CRC_NG` is set

Reply:
Yup, sounds good to me, this way we keep as much logic as possible in the unit file instead of having to go look at another script.

Unrelated thought, but I wonder if instead of running `crc-ng-env.sh` again and again in an `ExecCondition`, we could have a single unit file doing it, and then have the other units "depend" on that unit to only be started if the `crc-ng-env.sh` succeeded (or did not succeed)

Reply:
> maybe we should have two versions of this service, one with

> ConditionVirtualization=apple and the other with ExecCondition=/usr/local/bin/check-crc-ng-env.sh

> the first removes this file in case we are running on macOS, and the other service will remove it when CRC_CLOUD or CRC_NG is set

if we have crc-no-tap logic where it remove the tap device any of the condition run then why we want to have 2 versions of it? Does it fails for some condition?

Reply:
> Unrelated thought, but I wonder if instead of running crc-ng-env.sh again and again in an ExecCondition, we could have a single unit file doing it, and then have the other units "depend" on that unit to only be started if the crc-ng-env.sh succeeded (or did not succeed). Just food for thoughts, no need to address it in this PR.

yeah, something can be explored.

@coderabbitai create follow up issue

Reply:
> I prefer to have as much as possible in systemd units, and as little as possible in external shell scripts, this way everything is in one place.

Can you check again, I think now that external script not only delete tap configuration but also disable the forwarder service, do you still prefer to have 2 separate service doing same thing depending on the environment and virtualization?